main/main.cpp

```diff
@@ -1372,6 +1372,8 @@ bool Main::start() {
 #ifdef TOOLS_ENABLED
 	if (doc_tool != "") {
 
+		Engine::get_singleton()->set_editor_hint(true); // Needed to instance editor-only classes for their default values
```

Comment:
Won't this add various editor-only classes to the docs? See e.g. my changes in https://github.com/godotengine/godot/pull/29872/files#diff-7c557c73ffdf760d458e6226f4d70c31R43 where I had to check the editor hint to prevent registering `ResourceImporterOGGVorbis`.

Reply:
Tested it, and it doesn't seem to add `ResourceImporterOGGVorbis` back.

Reply:
This is more about EditorFileDialog and the like, since ClassDB does not allow instantiating them without the editor hint.